### PR TITLE
Rename main cmake/CMakeLists.txt to CMakeLists.txt

### PR DESCRIPTION
**Pull request by Arfrever**
cmake/CMakeLists.txt is kept for compatibility and it includes CMakeLists.txt.

Fixes: #9596

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,351 @@
+# Minimum CMake required
```
Comment (acozzette) [inline CMakeLists.txt:1]: Could you do a `git mv` from the old location so that we can preserve the history of this file?

Comment (Arfrever) [reply CMakeLists.txt:1]: I actually did `git mv cmake/CMakeLists.txt CMakeLists.txt` and later separate `git add cmake/CMakeLists.txt` for new file.
The only way to do what you want seems to be to have 2 separate commits.
Should I do it?

Comment (acozzette) [reply CMakeLists.txt:1]: Actually never mind what I said. From reading some more I think I misunderstood how Git handles file moves and copies. If I understand correctly now, Git doesn't try to express file moves through metadata but instead just tries to detect them after the fact. It's too bad the GitHub UI doesn't show the diff better, though.

Comment (Arfrever) [reply CMakeLists.txt:1]: I have split those changes into 2 commits, and added a clean-up commit (`protobuf_source_dir` -> `protobuf_SOURCE_DIR`), so if you merge these changes as 3 separate commits (not squashed), then history should be better shown.

Comment (acozzette) [reply CMakeLists.txt:1]: Awesome! I will take a closer look soon, but my only major concern so far is I think this change is a bit too big to try to get into the 3.20.0 release given that the release is already in progress. Would you mind rebasing this pull request to be against the master branch instead? I can first merge 3.20.x into master if that will help.

Comment (Arfrever) [reply CMakeLists.txt:1]: OK. Please merge `3.20.x` into `master`, and I will re-base these changes against `master`...

Comment (acozzette) [reply CMakeLists.txt:1]: Ok, 3.20.x has now been merged into master.

**Comment by acozzette** [comment]
Thanks, @Arfrever!

**Pull request merged**